<launch>
<arg name="base_frame" default="sensor"/>
<arg name="sensor_range" default="20.0"/>
<arg name="map_resolution" default="0.4"/>
<arg name="node_resolution" default="2.0"/>

  <node pkg="octomap_server" type="octomap_server_node" name="octomap" output="screen">
    <remap from="cloud_in" to="sensor_scan"/>
    <param name="frame_id" value="map"/>
    <param name="base_frame_id" value="$(arg base_frame)"/>
    <param name="resolution" value="$(arg map_resolution)"/>
    <param name="occupancy_min_z" type="double" value="0.0"/>
    <param name="occupancy_max_z" type="double" value="1.2"/>
    <param name="sensor_model/max_range" value="$(arg sensor_range)" />
    <param name="sensor_model/hit" value="1"/>
    <param name="sensor_model/miss" value="0.45"/>
    <param name="sensor_model/max" value="1"/>
    <param name="sensor_model/min" value="0.2"/>
  </node>

  <node pkg="rl_planner" type="rl_planner.py" name="rl_planner" output="screen" cwd="node">
    <param name="publish_graph" type="bool" value="true"/>
    <param name="node_resolution" value="$(arg node_resolution)"/>
    <param name="sensor_range" value="$(arg sensor_range)"/>
    <param name="utility_range_factor" value="0.5"/>
    <param name="min_utility" type="int" value="3"/>
    <param name="frontier_downsample_factor" type="int" value="1"/>
    <param name="map_resolution" value="$(arg map_resolution)"/>
    <param name="waypoint_threshold" value="2.0"/>
    <param name="next_waypoint_threshold" value="4.0"/>
    <param name="hard_update_threshold" value="10.0"/>
    <param name="frontier_sample_factor" value="10.0"/>
    <param name="replanning_frequency" value="2.5"/>
  </node>

  <node pkg="tf" type="static_transform_publisher" name="link_broadcaster" args="0 0 0 0 0 0 1 vehicle base_link 100" />

  <node launch-prefix="nice" pkg="rviz" type="rviz" name="rl_rviz" args="-d $(find rl_planner)/rviz/rviz.rviz" respawn="true"/>
</launch>
